src/app.rs

```diff
@@ -307,10 +308,120 @@ impl App {
             }
             KeyCode::Esc => self.reset_message_selection(),
             KeyCode::Char(c) => self.put_char(c),
+            KeyCode::Tab => {
+                if let Some(idx) = self.data.channels.state.selected() {
+                    self.add_reaction(idx, self.data.input.is_empty()).await;
+                }
+            }
             _ => {}
         }
     }
 
+    pub async fn add_reaction(&mut self, channel_idx: usize, remove: bool) {
+        let emoji_replacer = Replacer::new();
+
+        let input: String = self.data.input.drain(..).collect();
+        let reaction: String = emoji_replacer.replace_all(&input).to_string();
```

Comment:
Maybe we could use something like that to try to take the emoji out of the input

```rust

/// Returns the emoji and leaves the `input` empty if it is of the shape `:some_real_emoji`.
fn take_emoji(input: &mut String) -> Option<String> {
    let s = input.trim();
    let s = s.strip_prefix(':')?.strip_suffix(':')?;
    let emoji = gh_emoji::get(s)?.to_string();
    input.clear();
    Some(emoji)
}
```

In that case, we would not react with non existent emojis.

Reply:
I thought of something similar but this would reject a string consisting of a emoji character (e.g. :smile: and not `:smile:`)

Reply:
Maybe this helps? https://docs.rs/unic-emoji-char/0.9.0/unic_emoji_char/fn.is_emoji.html

Reply:
This will probably do the trick. I only have one concern : this method only takes in a single character. However, IIRC, some emojis (like the ones with skin tone variations) consist of multiple characters. This means such emojis would probably be rejected. But maybe not a huge deal at first!

Reply:
Good point. Then maybe this:

```rust
emoji::lookup_by_glyph::lookup("🚴🏿")).is_some()
```

Reply:
I just pushed a commit with this function as emoji detection mechanism